### PR TITLE
LF-4390: Persist error snackbars

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

- Implement the ability to persist the snackbar.
- Remove animation for persisted snackbars.
- Update `enqueueErrorSnackbar` reducer to persist all error snackbars. (The decision was made in [JIRA ticket](https://lite-farm.atlassian.net/browse/LF-4390?focusedCommentId=17723))
- Update [story](http://localhost:6006/?path=/story/components-snackbar--primary)

There were some issues implementing the persisted snackbar:
1. `persist` was hard-coded to true in `useReduxSnackbar`.
2. The snackbar was always being dismissed in `packages/webapp/src/components/Map/ProgressBar/index.jsx`. (This was overriding `{ persist: true }` in `useReduxSnackbar`)
3. Only `key` and `message` could be passed as props to `NotistackSnackbar` (it wasn't possible to conditionally dismiss the snackbar).

The solutions are as follows:
1. `persist: true` is now passed as argument to the `enqueueSnackbar` function called in `useReduxSnackbar`.
2. `onDismiss` prop is now conditionally passed to `ProgressBar` component.
3. `NotistackSnackbar` is rendered as the `SnackbarProvider`'s `Components` prop. ([documentation](https://notistack.com/features/customization#custom-component)) The `Components` object’s keys are `variant`, and its values are components. Since we have only one snackbar, I named the variant "common". The variant is passed as an argument to `enqueueSnackbar`.

Jira link: https://lite-farm.atlassian.net/browse/LF-4390

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

To simulate an error, I added `throw Error("testing")` in the `AddAnimals` function in `packages/webapp/src/containers/Animals/AddAnimals/index.tsx` and `createTaskSaga` in `packages/webapp/src/containers/Task/saga.js`.

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Comment by SayakaOno** [comment]
Sorry, I'm confirming one more thing with Loïc...

File: packages/webapp/src/components/Map/ProgressBar/index.jsx [hunk 1]
```diff
@@ -37,7 +41,7 @@ export default function ProgressBar({ onDismiss, type = 'success' }) {
     return () => {
       clearInterval(timer);
     };
-  }, []);
+  }, [onDismiss]);
 
   useEffect(() => {
     if (progress >= MAX_PROGRESS) {
```
Comment (SayakaOno) [inline packages/webapp/src/components/Map/ProgressBar/index.jsx:47]: This component was always receiving `onDismiss` prop, so `onDismiss?.()` (on the next line) was always being called.
Now, `onDismiss` is conditionally passed from the parent component.

**Review by antsgar — APPROVED** [review]
Great job @SayakaOno!! 🔥

**Comment by kathyavini** [comment]
This is awesome!!

Maybe we can keep an special eye out now for those random error snackbars that sometimes trigger on logging into some farms? With the persistence users might be confused into feeling they have to act on them in some way, so hopefully we can just spot them and clean them up before users ever see it on prod! (They were kind of annoying anyway to start!)

**Comment by SayakaOno** [comment]
Thank you @kathyavini for the review and for bringing that up!
Is there a ticket for it? I'd love to have it fixed before the release!!

**Comment by kathyavini** [comment]
@SayakaOno I don't believe a ticket was ever made (or at least I can't track one down) probably because it's so intermittent and then the snackbar (had) just passed so you don't worry about it much! 😂 I have a feeling we should run into it again soon, and I'll give Denis a heads up too.

**Pull request merged**